prepline_general/api/general.py

```diff
@@ -58,28 +59,115 @@ def is_expected_response_type(media_type, response_type):
     os.environ["UNSTRUCTURED_ALLOWED_MIMETYPES"] = DEFAULT_MIMETYPES
 
 
+def get_pdf_splits(pdf, split_size=1):
+    """
+    Given a pdf (PdfReader) with n pages, split it into pdfs each with split_size # of pages
+    Return the files as a list of BytesIO
+    """
+    split_pdfs = []
+
+    count = 0
+
+    while count < len(pdf.pages):
+        new_pdf = PdfWriter()
+        pdf_buffer = io.BytesIO()
+
+        end = count + split_size
+        for page in pdf.pages[count:end]:
+            new_pdf.add_page(page)
+
+        new_pdf.write(pdf_buffer)
+        pdf_buffer.seek(0)
+
+        split_pdfs.append(pdf_buffer)
+        count += split_size
+
+    return split_pdfs
+
+
+def partition_file_via_api(file, **kwargs):
+    """
+    Given a file-like, use partition_via_api to retrieve its elements.
+    Hardcode sending it to api.unstructured.io for now.
+    """
+    # Note (austin) - the correct thing is to call back to ourselves
+    # using the request.url. We'll need to pass the Request into pipeline_api().
+    # Same goes for any request context to pass on, i.e. the token
+    request_url = "https://api.unstructured.io/general/v0/general"
```

Comment:
any reason not to do that now? some unexpected complexity?

Reply:
Mostly just that it's an api-tools change. We'll be back in here to figure out how to improve the speedup, so I figured I'd get something out to start.

Reply:
the request_url should:

1/ be an env var
2/ not be the external DNS name in the first place (i.e., you probably don't want to route through the internet)

Reply:
Ah, good point! I had also been wondering if it's worth having a private load balancer for this traffic, but that's perhaps too much infra work right now.